src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -231,21 +231,27 @@ HashSet<ITypeSymbol> typeStack
 			}
 		}
 
-		private MutabilityInspectionResult InspectBaseType(
+		public MutabilityInspectionResult InspectConcreteType(
 			ITypeSymbol type,
-			HashSet<ITypeSymbol> typeStack
+			MutabilityInspectionFlags flags
 		) {
-			if( type.BaseType == null ) {
-				return MutabilityInspectionResult.NotMutable();
-			}
+			var typesInCurrentCycle = new HashSet<ITypeSymbol>();
 
-			if( type.BaseType is IErrorTypeSymbol ) {
+			return InspectConcreteType( type, typesInCurrentCycle, flags );
+		}
+
+		private MutabilityInspectionResult InspectConcreteType(
+			ITypeSymbol type,
+			HashSet<ITypeSymbol> typeStack,
+			MutabilityInspectionFlags flags = MutabilityInspectionFlags.Default

```

Comment:
This exists to support ImmutabilityAnalyzer passing `IgnoreImmutable`. I want to kill that flag too but that's a bit more work that makes sense after a bunch of other changes. The default means that internally things look mostly sane.

Reply:
Might be nice at some point to make the private recursive ones named differently, like `InspectConcreteTypeRecursive`.